casual/casual-network/src/main/java/se/laz/casual/network/outbound/NettyNetworkConnection.java

```diff
@@ -72,10 +81,13 @@ public static NetworkConnection of(final NettyConnectionInformation ci, final Ne
         NettyNetworkConnection networkConnection = new NettyNetworkConnection(ci, correlator, ch, conversationMessageStorage, EventLoopFactory.getManagedExecutorService());
         LOG.finest(() -> networkConnection + " connected to: " + ci.getAddress());
         ch.closeFuture().addListener(f -> handleClose(networkConnection, networkListener));
-        networkConnection.throwIfProtocolVersionNotSupportedByEIS(ci.getProtocolVersion(), ci.getDomainId(), ci.getDomainName());
+        DomainId id = networkConnection.throwIfProtocolVersionNotSupportedByEIS(ci.getProtocolVersion(), ci.getDomainId(), ci.getDomainName());
```

Comment:
I am confused, `ci.getDomainId()`, is that not the same thing as we are being returned?

Reply:
The domain id is part of the handshake reply.

Reply:
One could argue that the method should be renamed to something like handshake perhaps?

Reply:
> The domain id is part of the handshake reply.

So what we can ask to connect to a domain id and another domain replies?

Reply:
Each real network connetion only connects to one domain, during the handshake we get the domain id back in the reply.
For a pool of managed connections going to the same host:port it may be that the domain id for each managed connection is different. It can also be the same domain id - we do not know which.

Reply:
The problem we have today in such an application, casual caller for instance, is that it is not granular enough since it is caching based on connection factory jdni-name, that is - it expects all connections in a pool to be connected to the same domain - that is not necessarily true.